firebase/functions/src/index.ts

```diff
@@ -179,6 +186,33 @@ export const createChain = functions
           chainId: chainData.id,
           role: user.customClaims?.role ?? ROLE_CHAINADMIN,
         });
+
+        try {
+          const name = user.displayName!;
+          const email = user.email!;
+
+          const nameLength = name.split(" ").length;
+          const firstName =
+          nameLength === 1 ? name : name.split(" ").slice(0, -1).join(" ");
+          const lastName = name.split(" ").slice(-1).join(" ");
+
+          await mailchimp.lists.addListMember(
+              functions.config().mailchimp.interested_audience_id,
+              {
+                email_address: email,
+                status: "subscribed",
+                merge_fields: {
+                  FNAME: firstName,
+                  LNAME: lastName,
+                },
+              }
+          );
+        } catch (error) {
+          console.log("Mailchimp add contact error", error);
+        }
+
+        console.log("Mailchimp add contact success");
```

Comment:
Might be good to add a separate function for this bit.

Reply:
Also, move the function inside the try block. Otherwise on error you get "Mailchimp add contact error", "Mailchimp add contact success".